cynghanedd.py

```diff
@@ -194,9 +194,20 @@ def choose_word(solutions, word):
     remaining_solutions = [ solution[1:] for solution in solutions if solution[0] == skeleton ]
     return remaining_solutions
 
-if __name__ == '__main__':
-    import sys
+def complete_cyng(words, skeleton, solution):
```

Comment:
It's not completely clear to me what these three functions (complete_cyng, run_cynghanedd and back_cynghanedd) each do. It might be helpful to try to write a short description in English of what each does as a comment above it.